pkg/downloader/manager.go

```diff
@@ -659,10 +659,28 @@ func (m *Manager) UpdateRepositories() error {
 	return nil
 }
 
+// Filter out duplicate repos by URL, including those with trailing slashes.
+func dedupeRepos(repos []*repo.Entry) []*repo.Entry {
+	seen := make(map[string]*repo.Entry)
+	for _, r := range repos {
+		// Normalize URL by removing trailing slashes.
+		r.URL = strings.TrimSuffix(r.URL, "/")
```

Comment:
I don't think this normalization should be done here (it is updating the original repo object via the pointer).

Preferably IMHO, the chart repo's URL would be normalized when the repo is loaded.

A more hacky way, the "seen" map could be keyed by a normalized URL. But the passed `repos` not mutated.